src/Exception/AuthenticationException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Joindin\Api\Exception;
+
+use \Exception;
+use \Throwable;
+
+final class AuthenticationException extends Exception implements Throwable
```

Comment:
Perhaps it makes sense to extend a more specific exception, for example,  `RuntimeException`?